main.go

```diff
@@ -60,25 +60,20 @@ func main() {
 		printErrorFatal("Invalid Usage", err)
 	}
 
-	if len(ctx.Args) > 0 {
-		switch ctx.Args[0] {
-
-		case "version":
-			if Version == "" {
-				if info, ok := debug.ReadBuildInfo(); ok && info.Main.Sum != "" {
-					Version = info.Main.Version
-				} else {
-					Version = "unknown (built from source)"
-				}
-			}
-			version := fmt.Sprintf("freeze version %s", Version)
-			if len(CommitSHA) >= shaLen {
-				version += " (" + CommitSHA[:shaLen] + ")"
+	if config.Version {
+		if Version == "" {
+			if info, ok := debug.ReadBuildInfo(); ok && (info.Main.Version != "" && info.Main.Version != "(devel)") {
+				Version = info.Main.Version
+			} else {
+				Version = "latest (built from source)"
 			}
-
-			fmt.Println(version)
-			os.Exit(0)
 		}
+		version := fmt.Sprintf("freeze version: %s", Version)
+		if len(CommitSHA) > 7 {
+			version += fmt.Sprintf(" (%s)", CommitSHA[:shaLen])
```

Comment:
So the sha is jot displayed if the sha is shorter than 7 ?

Reply:
@ccoVeille I made the changes you proposed, added a comment about `CommitSHA`, I meant to use `shaLen` instead of 7 but I forgot, my bad on that.

What it does is that if the `CommitSHA` is greater than `shaLen` at this moment 7, it will print the first 7 characters of `CommitSHA` instead of the full SHA for the commit.

Reply:
I'm glad my review helped you